controllers/windowsmachine_controller.go

```diff
@@ -39,41 +38,45 @@ import (
 )
 
 const (
-	// WindowsMachineControllerName is the name of the WindowsMachine controller
-	WindowsMachineControllerName = "windowsmachine-controller"
 	// maxUnhealthyCount is the maximum number of nodes that are not ready to serve at a given time.
 	// TODO: https://issues.redhat.com/browse/WINC-524
 	maxUnhealthyCount = 1
 	// MachineOSLabel is the label used to identify the Windows Machines.
 	MachineOSLabel = "machine.openshift.io/os-id"
 )
 
-// AddWindowsMachineController creates a new WindowsMachine Controller and adds it to the Manager. The Manager will set fields on the Controller
-// and start it when the Manager is Started.
-func AddWindowsMachineController(mgr manager.Manager, clusterConfig cluster.Config, watchNamespace string) error {
-	reconciler, err := newWindowsMachineReconciler(mgr, clusterConfig, watchNamespace)
-	if err != nil {
-		return errors.Wrapf(err, "could not create %s reconciler", WindowsMachineControllerName)
-	}
-	return add(mgr, reconciler)
+// WindowsMachineReconciler is used to create a controller which manages Windows Machine objects
+type WindowsMachineReconciler struct {
+	client client.Client
+	log    logr.Logger
+	scheme *runtime.Scheme
+	// k8sclientset holds the kube client that we can re-use for all kube objects other than custom resources.
+	k8sclientset *kubernetes.Clientset
+	// clusterServiceCIDR holds the cluster network service CIDR
+	clusterServiceCIDR string
+	// signer is a signer created from the user's private key
+	signer ssh.Signer
+	// vxlanPort is the custom VXLAN port
+	vxlanPort string
+	// recorder to generate events
+	recorder record.EventRecorder
+	// watchNamespace is the namespace the operator is watching as defined by the operator CSV
+	watchNamespace string
+	// prometheusConfig stores information required to configure Prometheus
+	prometheusNodeConfig *metrics.PrometheusNodeConfig
+	// platform indicates the cloud on which OpenShift cluster is running
+	// TODO: Remove this once we figure out how to be provider agnostic. This is specific to proper usage of userData
+	// 		 in vSphere
+	//		https://bugzilla.redhat.com/show_bug.cgi?id=1876987
+	platform oconfig.PlatformType
 }
 
-// newWindowsMachineReconciler returns a new reconcile.Reconciler
-func newWindowsMachineReconciler(mgr manager.Manager, clusterConfig cluster.Config, watchNamespace string) (reconcile.Reconciler, error) {
+// NewWindowsMachineReconciler returns a pointer to a WindowsMachineReconciler
+func NewWindowsMachineReconciler(mgr manager.Manager, clusterConfig cluster.Config, watchNamespace string) (*WindowsMachineReconciler, error) {
 	// The default client serves read requests from the cache which
 	// could be stale and result in a get call to return an older version
 	// of the object. Hence we are using a non-default-client referenced
 	// by operator-sdk.
```

Comment:
is this comment section valid now?

Reply:
I cant say with certainty, i think follow up work should be done to see if the non-client clientset can be removed

Reply:
I don't think it is valid now, the mgr.GetClient() you are using is the deafult client and not a non default one as we create below using 
```
cfg, err := config.GetConfig()
client, err := client.New(cfg, client.Options{Scheme: mgr.GetScheme()})
```
Though I am wondering if we want to stick to using a default client or reuse the one we were already creating?

Reply:
Investigating this is taking a decent amount of work so I've created a spike that someone can pick up later: https://issues.redhat.com/browse/WINC-597

Reply:
I feel this might be related to the other discussion regarding the cache: https://github.com/openshift/windows-machine-config-operator/pull/341/commits/c5b54cd622436b49942c75826102a035bf0908a7#r608113020. I am not sure we can push this off to a spike without understanding what is going on. This could potentially lead to subtle bugs.